security-admin/src/test/java/org/apache/ranger/common/TestRangerConfigUtil.java

```diff
@@ -48,4 +48,22 @@ public void testGetRoles(){
 		String[] str=xaConfigUtil.getRoles();
 		Assert.assertArrayEquals(str, xaConfigUtil.roles);
 	}
+
+	@Test
+	public void testIsUserGroupManagementEnabled(){
+		boolean value = xaConfigUtil.isUserGroupManagementEnabled();
+		Assert.assertTrue(value);
+	}
+
+	@Test
+	public void testIsAdminDelegationEnabled(){
+		boolean value = xaConfigUtil.isAdminDelegationEnabled();
+		Assert.assertTrue(value);
+	}
+
+	@Test
+	public void testIsServiceManagementEnabled(){
+		boolean value = xaConfigUtil.isServiceManagementEnabled();
+		Assert.assertTrue(value);
+	}
```

Comment:
There were many changes and improvements to existing functionality, however only three tests were added. It would be a good idea to also test the middleware and REST API calls to make sure the restrictions are scoped properly. This would include but is not limited to unit, controller and/or integration tests.

Reply:
overall, looks good. Please take a pass and try to correct as many style issues as you can. We def. need more tests, however, at this point I would even first merge this and then add tests in next PR. Please make PRs smaller and merge more often. Because then it's easier to make progress and get meaningful review that can help you.

Either in this or in next pass, we need to focus on authenticate method and write more tests for it and refactor it to make it simpler. Or at least keep our changes as small as possible comparing to ranger mainline.

Reply:
Few more comments. I think we'll have to create a clean patch when moving this upstream to community. For now, We definitely want to start using RPM with these changes, so I will approve this PR with understanding we'll do another cleanup before contributing this to community.